mlflow/projects/databricks.py

```diff
@@ -162,12 +162,26 @@ def _run_shell_command_job(self, project_uri, command, env_vars, cluster_spec):
         :param command: Shell command to run.
         :param env_vars: Environment variables to set in the process running ``command``.
         :param cluster_spec: Dictionary containing a `Databricks cluster specification
-                             <https://docs.databricks.com/api/latest/jobs.html#clusterspec>`_
-                             to use when launching a run.
+                             <https://docs.databricks.com/dev-tools/api/latest/jobs.html#clusterspec>`_
+                             or a `Databricks new cluster specification
+                             <https://docs.databricks.com/dev-tools/api/latest/jobs.html#jobsclusterspecnewcluster>`_
+                             to use when launching a run. If you specify libraries, this function
+                             will add MLflow to the library list. This function does not support
+                             installation of conda environment libraries on the workers.
         :return: ID of the Databricks job run. Can be used to query the run's status via the
                  Databricks
                  `Runs Get <https://docs.databricks.com/api/latest/jobs.html#runs-get>`_ API.
         """
+        # NB: We use <= on the version specifier to allow running projects on pre-release
+        # versions, where we will select the most up-to-date mlflow version available.
+        # Also note, that we escape this so '<' is not treated as a shell pipe.
+        libraries = [{"pypi": {"package": "'mlflow<=%s'" % VERSION}}]
+
+        # Check syntax of JSON - if it contains a libraries and a cluster_spec, pull those out
+        if 'libraries' in cluster_spec and 'new_cluster' in cluster_spec:
```

Comment:
Given that we document that a `new_cluster_spec` with [these fields](https://docs.databricks.com/dev-tools/api/latest/jobs.html#clusterspec) is valid, I suspect that we may want to allow specifying a `new_cluster` without libraries, e.g via logic like:

```suggestion
    if 'new_cluster' in cluster_spec:
        if 'libraries' in cluster_spec:
            libraries.extend(cluster_spec['libraries'])
        cluster_spec = cluster_spec['new_cluster']
```

We'd also then want a unit test similar to [the one you already added](https://github.com/mlflow/mlflow/pull/2845/files#diff-e5f3a2fbd5bd330b3069ea935f1718eeR273-R296) for specifying `new_cluster` without `libraries`. I'm open to adding this support in a follow-up PR - if we go that route, we should probably raise an exception if the user specifies `new_cluster` but no `libraries`, as we otherwise currently just pass an object like `{"new_cluster": {...}}` to the REST API, which I think will fail to run the job. Let me know what you think, thanks!

Reply:
Thanks, @smurching! I've incorporated your feedback to make `libraries` optional in the specification and added a test covering that.